### PR TITLE
Feature/lti service abstraction layer

### PR DESCRIPTION
**Pull request by ekkinox**
**Details**
* Added github actions CI
* Added LTI service layer automation to ease LTI libraries usage through the bundle
* Removed jenkins and travis CI
* Updated oat-sa/lib-lti1p3-core dependency to version 6.0
* Updated documentation

**Additional info**
* related documentation: https://github.com/oat-sa/bundle-lti1p3/blob/feature/lti-service-abstraction-layer/doc/service/platform.md#providing-platform-service-endpoints-using-the-lti-libraries
* target release: `6.0.0`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 CHANGELOG
 =========
 
+6.0.0
+-----
+
+* Updated oat-sa/lib-lti1p3-core dependency to version 6.0
+* Added LTI service layer automation to ease LTI libraries usage through the bundle  
+* Updated documentation
```
Comment (m-lagarde) [inline CHANGELOG.md:11]: Can we group `updated` together? :slightly_smiling_face:

Comment (ekkinox) [reply CHANGELOG.md:11]: sure ! done.

File: Service/Server/Factory/LtiServiceServerHttpFoundationRequestHandlerFactory.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    public function __construct(
+        Security $security,
+        HttpFoundationFactoryInterface $httpFoundationFactory,
+        HttpMessageFactoryInterface $psr7Factory,
+        LoggerInterface $logger
+    ) {
+        $this->security = $security;
+        $this->httpFoundationFactory = $httpFoundationFactory;
+        $this->psr7Factory = $psr7Factory;
```
Comment (GregoryGagua) [inline Service/Server/Factory/LtiServiceServerHttpFoundationRequestHandlerFactory.php:55]: Maybe $httpMessageFactory would be more suitable here?

File: Service/Server/Handler/LtiServiceServerHttpFoundationRequestHandler.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        Security $security,
+        HttpFoundationFactoryInterface $httpFoundationFactory,
+        HttpMessageFactoryInterface $psr7Factory,
+        LoggerInterface $logger,
+        array $options = []
+    ) {
+        $this->handler = $handler;
+        $this->security = $security;
+        $this->httpFoundationFactory = $httpFoundationFactory;
+        $this->psr7Factory = $psr7Factory;
```
Comment (GregoryGagua) [inline Service/Server/Handler/LtiServiceServerHttpFoundationRequestHandler.php:68]: Same question.

File: Service/Server/Factory/LtiServiceServerHttpFoundationRequestHandlerFactory.php (hunk 2 above)
Comment (ekkinox) [reply Service/Server/Factory/LtiServiceServerHttpFoundationRequestHandlerFactory.php:55]: it's alrfeady named this way in many places, imo offers better distinction with http foundation, and it's to follow SF doc: https://symfony.com/doc/current/components/psr7.html#usage

if you're ok I'd like to keep it this way

File: Service/Server/Handler/LtiServiceServerHttpFoundationRequestHandlerInterface.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+declare(strict_types=1);
+
+namespace OAT\Bundle\Lti1p3Bundle\Service\Server\Handler;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+interface LtiServiceServerHttpFoundationRequestHandlerInterface
+{
+    public function __invoke(Request $request): Response;
```
Comment (GregoryGagua) [inline Service/Server/Handler/LtiServiceServerHttpFoundationRequestHandlerInterface.php:34]: Missing @throws annotation

File: Service/Server/Factory/LtiServiceServerHttpFoundationRequestHandlerFactory.php (hunk 2 above)
Comment (GregoryGagua) [reply Service/Server/Factory/LtiServiceServerHttpFoundationRequestHandlerFactory.php:55]: Sure, I don't insist here.

File: Service/Server/Handler/LtiServiceServerHttpFoundationRequestHandler.php (hunk 3 above)
Comment (ekkinox) [reply Service/Server/Handler/LtiServiceServerHttpFoundationRequestHandler.php:68]: https://github.com/oat-sa/bundle-lti1p3/pull/40#discussion_r613897575

File: Service/Server/Handler/LtiServiceServerHttpFoundationRequestHandlerInterface.php (hunk 4 above)
Comment (ekkinox) [reply Service/Server/Handler/LtiServiceServerHttpFoundationRequestHandlerInterface.php:34]: I add it, just for you, but you know my opinion on this :)

**Review by GregoryGagua — APPROVED** [review]

**Review by m-lagarde — APPROVED** [review]

**Pull request merged**